ls1intum/Artemis
# Improve server code and test quality

* Optimize imports with default IntelliJ settings.
* Use exceptions directly in AuthorizationCheckService instead of `forbidden()` for security checks in Resources to reduce code
* Use `findByIdWithXXXElseThrow()` in Resources to reduce 
* Remove the deprecated method `findWithAdditionalElements()` 
* Improve test quality by using `private` consistently and by removing unused dependencies
* Fix many wrong usages of `assertThat()` --> we should really be careful here, otherwise the assertions are completely useless
* Improve the implementation of the role concept
* Add a role hierarchy to simplify security checks and make them more consistent

### Checklist

### Steps for Testing
* The complete quiz exercise lifecycle (create, edit, participate, re-evaluate, check **statistics**, delete)
* The complete programming exercise lifecycle (create, edit, participate, delete) including grading and manual assessment.
